### PR TITLE
Add test helpers

### PR DESCRIPTION
**Pull request by devpow112**
Adds helper for eliminating resize observer loops and adds the new HTML editor require context to all demo pages and tests. This helps with stability and error message output during vdiff and unit/axe tests.

### CODE REVIEW

File: wds.config.js [hunk 1]
```diff
@@ -15,17 +15,49 @@ export const devServerConfig = {
...
+									const featureValue = window.D2L.LP.Web.UI.Flags.ListedFlags[feature];
+
+									return featureValue !== undefined ? featureValue : defaultValue;
+								}
+							};
+
+							document.documentElement.dataset.mathjaxContext = JSON.stringify({
+								outputScale: 1.1,
+								renderLatex: true
+							});
```
Comment (devpow112) [inline wds.config.js:44]: Gonna look at porting this into the htmleditor repo so we can use it in other places if needed.

File: wds.config.js [hunk 2]
```diff
@@ -15,17 +15,32 @@ export const devServerConfig = {
...
 			}
-		})
+		}),
+		{
+			name: 'setup-html-editor-context',
+			transform(context) {
+				if (context.response.is('html')) {
+					const newBody = context.body.replace(
+						/<head>/,
+						'<head><script src="/tools/htmleditor-mathjax-test-context.js"></script>'
```
Comment (devpow112) [inline wds.config.js:25]: Auto add HTML editor context to all demo and test pages.

File: wds.config.js [hunk 3]
```diff
@@ -15,17 +15,32 @@ export const devServerConfig = {
...
 	devServerConfig,
 	{
 		plugins: [{
-			name: 'stub-post',
+			name: 'stub-actions',
 			serve(context) {
-				if (context.request.method === 'POST') {
+				const method = context?.request?.method?.toLowerCase();
+
+				if (['put', 'post'].includes(method)) {
```
Comment (devpow112) [inline wds.config.js:43]: Also stub `PUT` requests in demo pages.

File: test/perceptual/helpers/page.js [hunk 4]
```diff
@@ -32,13 +30,6 @@ export const openCompactDemoPageElements = (tagNames) => {
 			tagNames = [tagNames];
 		}
 
-		const script = document.createElement('script');
```
Comment (devpow112) [inline test/perceptual/helpers/page.js:35]: This loading was too flaky. Just adding it to each page.

File: test/perceptual/helpers/page.js [hunk 5]
```diff
@@ -60,10 +51,7 @@ export const setUpFeatureFlags = () => {
 	window.D2L.LP.Web = window.D2L.LP.Web || {};
 	window.D2L.LP.Web.UI = window.D2L.LP.Web.UI || {};
 	window.D2L.LP.Web.UI.Flags = window.D2L.LP.Web.UI.Flags || {
-		ListedFlags: {
-			'rubrics-lit': true,
-			'us125413-mathjax-render-latex': true
```
Comment (devpow112) [inline test/perceptual/helpers/page.js:65]: Not needed anymore

File: wds.config.js (hunk 2 above)
Comment (imcdonald-d2l) [reply wds.config.js:25]: Do we still need this line `<script src="/tools/htmleditor-mathjax-test-context.js"></script>` in the vdiff `html` files then?

Comment (devpow112) [reply wds.config.js:25]: Sadly yes. This snippet is only run when you run `npm run start:vdiff` (basically when the `@web/dev-server` or `@web/test-runner` are run) when vdiff actually runs it doesn't use any automated setup.

Comment (imcdonald-d2l) [reply wds.config.js:25]: Ah makes sense, thanks for clarifying.

Comment (devpow112) [reply wds.config.js:25]: I'm hoping to look at adding something like this to vdiff in maybe the next inspiration/dragon slayer

**Review by imcdonald-d2l — APPROVED** [review]

**Pull request merged**